bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+messages: nodes are allowed to send at most one announcement and update per
+block. To allow for bursts, nodes are encouraged not to use the latest block
+height for their latest announcements/updates but rather to backdate and use
+older block heights that they have not used in an announcement/update. There of
+course needs to be a limit on the start block height that the node can use:
+for `channel_update_2` messages, the first timestamp must be the block height in
+which the channel funding transaction was mined and all updates after the
+initial one must have increasing timestamps. Nodes are then responsible for
+building up their own timestamp buffer: if they want to be able to send
+multiple `channel_update_2` messages per block, then they will need to ensure 
+that there are blocks during which they do not broadcast any updates. This 
+provides an incentive for nodes not to spam the network with too many updates.
+
+To also prevent nodes from building up too large of a burst-buffer with which 
+they can spam the network and to give a limit to how low the block height on a 
+`node_announcement_2` can be: nodes should not be allowed to use a block height
+smaller 1440 (~ a day worth of blocks) below the current block height.
+
+### Simplifies Channel Announcement Queries
+
+In the legacy gossip protocol, the timestamp of the `channel_announcement` is
+hard to define since the message itself does not have a `timestamp` field. This
+makes timestamp based gossip queries tricky. By using block heights as 
+timestamps instead, there is an implicit timestamp associated with the 
+`channel_announcement_2`: the block in which the funding transaction is mined.
+
+## Bootstrapping Taproot Gossip
+
+While the network upgrades from the legacy gossip protocol to the taproot gossip
+protocol, the following scenarios may exist for any node on the network:
+
+| scenario | has legacy channels | has taproot channels  | should send `node_announcement` | should send `node_announcement_2` |
+|----------|---------------------|-----------------------|---------------------------------|-----------------------------------|
+| 1        | no                  | no                    | no                              | no                                |
+| 2        | yes                 | no                    | yes                             | no                                |
+| 3        | yes                 | yes                   | yes                             | no                                |
+| 4        | no                  | yes                   | no                              | yes                               |
+
+### Scenario 1
+
+These nodes have no announced channels and so should not be broadcasting legacy
+or new node announcement messages.
+
+### Scenario 2
+
+If a node has legacy channels but no taproot channels, they should broadcast
+only a legacy `node_announcement` message. Both taproot-gossip aware and unaware
```

Comment:
👍 

Will be important to make sure things are properly bridged before taproot chans+gossip are deployed in a widespread manner. Nodes may also want to make more "feeler" connections than usual if most of the direct channel peers haven't yet updated as well.

Reply:
Also the various systems deployed for allowing nodes to sync the gossip data over HTTP or w/e will help here as well. Not sure if any of them also allow nodes/users to _post_ gossip data as well though.